### PR TITLE
QVM weblet

### LINKED ISSUE #126: QVM weblet
QVM short for QSFS enabled virtual machine

the idea should be a small machine 
- ubuntu based with ssh enabled
- 1 CPU 
- 2 GB RAM
- should have 10 ZDBs provisioned ? farm distribution?
- QSFS 
- Should be able to see the metrics / insights of the QVM on some endpoint

## Testing 
- system needs to work
- should withstand failures/repair
- should be able to see the metrics

### PR DESCRIPTION
**Pull request by 0oM4R**
### Description
As in threefoldtech/tfgrid-sdk-ts#126, VM with QSFS solution added to weblet

### Changes

- Add QVM  to 

> 1. App.vue
> 2. editor.vue
> 3. App.svelte
> 4. DeployedList.wc.svelte
> 5. currentDeployment.ts _with QSFS_
> 6. solutionsList.ts

- Create QSFS type 
- Create Qvm element
- Create DeployQvm
- Add QSFS to DeployVM.ts
- Add qsfsSisk to Vm type
- Add some validation in validateName.ts
- Add delete qsfs in deleteDeployment.ts
- Add metrics label to Table.svelte to use it in view qsfs metrics 
- use Multiselect element in selectNodeId.svelte
- modify MultiSelect.svelte layout
### Related Issues

List of related issues
threefoldtech/tfgrid-sdk-ts#126 
### Checklist

- [x] Build pass
- [x] Code format and docstrings

### CODE REVIEW

**Review by xmonader — CHANGES_REQUESTED** [review]
lgtm

**Review by xmonader — APPROVED** [review]

File: src/utils/deployQVM.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  const qsfs = new QSFSZDBSModel();
+  qsfs.name = name;
+  qsfs.count = count;
+  qsfs.node_ids = nodeIds;
+  qsfs.password = secret;
+  qsfs.disk_size = filters.hru;
+  const metadate = {
+    type: "QSFS",
+    name: name,
+    projectName: `QSFS of vm ${vm.name}`,
```
Comment (AhmedHanafy725) [inline src/utils/deployQVM.ts]: that's too long for the project name, can't it be `QVM`?

File: src/utils/validateName.ts [hunk 2]
```diff
@@ -269,3 +269,14 @@ export function validateAlgoStorage(value: string, net, type): string | void {
   if (+value > storage+200)
     return `Maximum Storage for this deployment is ${storage+200}.`;
 }
+
+export function validateQsfsSecret(value: string): string | void{
+  if (!value.length) return `zDB password is required`
```
Comment (AhmedHanafy725) [inline src/utils/validateName.ts]:
```suggestion
  if (!value.length) return `ZDB password is required`
```

File: src/utils/validateName.ts [hunk 3]
```diff
@@ -269,3 +269,14 @@ export function validateAlgoStorage(value: string, net, type): string | void {
...
+}
+
+export function validateZdbCount(value:number): string| void{
+  if (!NUM_REGEX.test(value.toString()) || isNaN(+value))
+  return "zDB count must be a valid number.";
+  else if(value < 3) return `Minimum allowed zDB is 3 zDBs`;
+
```
Comment (AhmedHanafy725) [inline src/utils/validateName.ts]:
```suggestion
  return "ZDB count must be a valid number.";
  else if(value < 3) return `Minimum allowed ZDB is 3 ZDBs`;
  
```

File: src/elements/DeployedList/DeployedList.wc.svelte [hunk 4]
```diff
@@ -456,6 +458,23 @@
...
+          label: "Show Details",
+          click: (_, i) => (infoToShow = rows[i].details),
+          disabled: () => removing !== null,
+          loading: (i) => removing === rows[i].name,
+        },
+        {
+          type: "warning",
+          label: "Metrics",
+          click: (_, i) => {
+            const domain = rows[i].details.mounts[0]["metricsEndpoint"];
```
Comment (AhmedHanafy725) [inline src/elements/DeployedList/DeployedList.wc.svelte]: this endpoint is only available on the planetary network, should it be exposed over the gateway? @xmonader

File: src/elements/qvm/Qvm.wc.svelte [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+      symbol: "count",
+      placeholder: "How many ZDBs needed?",
+      type: "number",
+      validator: validateZdbCount,
+      invalid: false,
+    },
+    {
+      label: "Disk size (MB) ",
+      symbol: "disk",
+      placeholder: "Memory of each ZDB in MB",
```
Comment (AhmedHanafy725) [inline src/elements/qvm/Qvm.wc.svelte]: this is confusing, is it disk or memory?

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]

File: src/elements/qvm/Qvm.wc.svelte (hunk 5 above)
Comment (0oM4R) [reply src/elements/qvm/Qvm.wc.svelte]: resolved to be "Disk size"

File: src/elements/DeployedList/DeployedList.wc.svelte (hunk 4 above)
Comment (AhmedHanafy725) [reply src/elements/DeployedList/DeployedList.wc.svelte]: let's remove this button for the metrics endpoint, and just keep it in the deployment details

File: src/elements/qvm/Qvm.wc.svelte [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+      symbol: "count",
+      placeholder: "How many ZDBs needed?",
+      type: "number",
+      validator: validateZdbCount,
+      invalid: false,
+    },
+    {
+      label: "Disk size (MB) ",
+      symbol: "disk",
+      placeholder: "Disk size of each ZDB in MB",
```
Comment (AhmedHanafy725) [inline src/elements/qvm/Qvm.wc.svelte]: disk size in all weblets is in GB, can you convert it to be consistent?

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**